cassandra/src/main/java/io/micronaut/configuration/cassandra/health/CassandraHealthIndicator.java

```diff
@@ -0,0 +1,154 @@
+            .column(COL_RELEASE_VERSION)
+            .column(COL_CLUSTER_NAME)
+            .column(COL_CQL_VERSION)
+            .column(COL_DATA_CENTER)
+            .column(COL_NATIVE_PROTOCOL_VERSION)
+            .column(COL_PARTITIONER)
+            .column(COL_RACK)
+            .build()
+            .setConsistencyLevel(ConsistencyLevel.LOCAL_ONE);
+
+    private static final Logger LOG = LoggerFactory.getLogger(CassandraHealthIndicator.class);
+
+    private final CqlSession cqlSession;
+
+    /**
+     * Default constructor.
+     *
+     * @param cqlSession The The cassandra {@link CqlSession} to query for details
+     */
+    public CassandraHealthIndicator(final CqlSession cqlSession) {
+        this.cqlSession = cqlSession;
+    }
+
+    @Override
+    protected Map<String, Object> getHealthInformation() {
+        Map<String, Object> detail = new LinkedHashMap<>();
+        Map<UUID, Node> nodes = cqlSession.getMetadata().getNodes();
+        detail.put("session", cqlSession.isClosed() ? "CLOSED" : "OPEN");
+        Optional<CqlIdentifier> opKeyspace = cqlSession.getKeyspace();
+        if (opKeyspace.isPresent()) {
+            detail.put("keyspace", opKeyspace.get());
+        }
+        detail.put("nodes_count", nodes.keySet().size());
+
+        Map<UUID, Map<String, Object>> nodesMap = new HashMap<>();
+        Map<NodeState, Integer> nodeStateMap = new EnumMap<>(NodeState.class);
+        boolean up = false;
+        int i = 0;
+        for (Map.Entry<UUID, Node> entry : nodes.entrySet()) {
+            UUID uuid = entry.getKey();
+            Node node = entry.getValue();
+            nodeStateMap.merge(node.getState(), 1, (a, b) -> a + b);
+            if (node.getState() == NodeState.UP) {
+                up = true;
+            }
+            if (i++ < 10) {
+                Map<String, Object> nodeMap = new HashMap<>();
+                Optional<InetSocketAddress> opBroadcastAddress = node.getBroadcastAddress();
+                if (opBroadcastAddress.isPresent()) {
+                    nodeMap.put("broadcast_address", opBroadcastAddress.get().getAddress());
+                }
+                nodeMap.put("state", node.getState());
+                nodeMap.put("distance", node.getDistance());
+                nodeMap.put("open_connections", node.getOpenConnections());
+                nodeMap.put("cassandra_version", node.getCassandraVersion());
+                nodeMap.put("datacenter", node.getDatacenter());
+                nodeMap.put("rack", node.getRack());
+                nodeMap.put("uptime_ms", node.getUpSinceMillis());
+                nodeMap.put("is_reconnecting", node.isReconnecting());
+                nodesMap.put(uuid, nodeMap);
+            }
+        }
+        detail.put("nodes_state", nodeStateMap);
+        if (nodesMap.size() > 0) {
+            detail.put("nodes", nodesMap);
+        }
+        healthStatus = up ? HealthStatus.UP : HealthStatus.DOWN;
```

Comment:
I personally would stop here as the rest is only repeating a bit the same sort of information. But I will let a Micronaut developer chime in and evaluate if it's acceptable to execute the query below and trigger a network roundtrip every time we invoke this method.

Reply:
I didn't understand this comment, must remove validation query execution and result info?

Reply:
Yes, my suggestion is to remove lines 133-149 (the query execution + result set processing).

Reply:
I can remove, but I think `{"cql_version": "3.4.4", "native_protocol_version": "4"}` informations can be useful.

Reply:
For `cql_version` I don't think so, it's useful only when connecting with CQLSH. 

For `native_protocol_version`, maybe; this column indicates the maximum protocol version supported for the node. I think we can live without it because 1) it doesn't mean that the driver is using the max version to connect and 2) you can infer the max protocol version from `Node.getCassandraVersion()`.

Also, you should keep in mind that each time you execute this query, the driver will pick a random node in the cluster. So the returned information may vary, and especially info like datacenter, rack, Cassandra version, CQL version, protocol version, may change frequently depending on which node is selected. This makes the returned information a bit unreliable.

Reply:
In your query, the only information that cannot change is: `cluster_name` and `partitioner`.

The partitioner is not very useful for end users.

The cluster name can be useful though. If you want to keep `cluster_name` you can do so without the query:

```
if (cqlSession.getMetadata().getClusterName().isPresent()) {
    String clusterName = cqlSession.getMetadata().getClusterName().get();
}
```

Reply:
fixed